### ISSUE TITLE
[Feature Request] Init of loss and metric module

### ISSUE THREAD

**Issue by illian01** [Nota-NetsPresso/netspresso-trainer#133]
- [x] I have searched to see if a similar issue already exists.

**Is there any feature that you would like to add?**  
- Loss and metric are initialized on every epochs.
- I think they don't need to be initialized frequently because they are not variable during train procedure.
```
for num_epoch in range(START_EPOCH_ZERO_OR_ONE, self.conf.training.epochs + START_EPOCH_ZERO_OR_ONE):
    self.timer.start_record(name=f'train_epoch_{num_epoch}')
    self.loss = build_losses(self.conf.model, ignore_index=self.ignore_index)
    self.metric = build_metrics(self.task, self.conf.model, ignore_index=self.ignore_index, num_classes=self.num_classes)
```

**Is there any suggestion (or solution) to solve this issue?**  
- Loss and metric can be initialized on `set_train` function.
- If there are values to be 0 after a epoch, we can add `reset` method for that.
